test/puppetlabs/puppetdb/query/ast_test.clj

```diff
@@ -0,0 +1,132 @@
+(ns puppetlabs.puppetdb.query.ast-test
+  "Tests the behavior of AST queries."
+  (:require
+   [clj-time.coerce :as coerce-time]
+   [clj-time.core :refer [now]]
+   [clojure.test :refer :all]
+   [puppetlabs.kitchensink.core :refer [dissoc-in]]
+   [puppetlabs.puppetdb.cli.services :as cli-svcs]
+   [puppetlabs.puppetdb.command :refer [enqueue-command]]
+   [puppetlabs.puppetdb.examples :as examples]
+   [puppetlabs.puppetdb.testutils :refer [block-until-results]]
+   [puppetlabs.puppetdb.testutils.cli :refer [get-catalogs get-factsets]]
+   [puppetlabs.puppetdb.testutils.services :as svc-utils :refer [*server*]]
+   [puppetlabs.trapperkeeper.app :refer [get-service]]))
+
+(deftest ast-queries
+  (svc-utils/with-single-quiet-pdb-instance
+    (let [dispatcher (get-service *server* :PuppetDBCommandDispatcher)
+          pdb (get-service *server* :PuppetDBServer)
+          query (fn [ast] (cli-svcs/query pdb :v4 ast {} set))]
+
+      (enqueue-command dispatcher :replace-facts 4
```

Comment:
Perhaps my comment above covers it, but I wanted as direct and lightweight an AST test as possible.